pkg/universal-cnf/config/removeClientInterface_test.go

```diff
@@ -0,0 +1,243 @@
+		},
+		"Unfound dstIpAddr in interfaces": {
+			interfaces2,
+			conn1,
+			routes1,
+			fmt.Errorf("client interface with dstIpAddr %s not found", conn1.Context.IpContext.DstIpAddr),
+		},
+	}
+
+	for testName, c := range cases {
+		t.Logf("Running test case: %s", testName)
+
+		ucnfBackend := &mocks.UniversalCNFBackend{}
+		uce := &UniversalCNFEndpoint{
+			backend:  ucnfBackend,
+			dpConfig: &vpp.ConfigData{Interfaces: c.interfaces, Routes: c.routes},
+		}
+
+		removeConfig, err := uce.removeClientInterface(c.conn)
+		if c.expectedError != nil {
+			assert.NotNil(t, err)
+		} else {
+			assert.Nil(t, err)
+			assert.NotNil(t, removeConfig)
+			assert.Equal(t, 1, len(removeConfig.Interfaces))
+		}
+	}
+}
+
+var interfaces1 = []*interfaces.Interface{
+	{
+		IpAddresses: []string{"192.168.22.2/30"},
+	},
+}
+
+var interfaces2 = []*interfaces.Interface{
+	{
+		IpAddresses: []string{"192.168.22.1/30", "192.168.22.0/30"},
+	},
+}
+
+var interfaces3 = []*interfaces.Interface{
+	{
+		IpAddresses: []string{"192.168.22.2/30", "192.168.22.1/30", "192.168.22.0/30"},
+	},
+}
+
+var routes1 = []*l3.Route{
+	{
+		NextHopAddr: "192.168.22.1",
+		DstNetwork:  "192.168.0.0/16",
+	},
+}
+
+var routes2 = []*l3.Route{
+	{
+		NextHopAddr: "192.168.22.2",
+		DstNetwork:  "192.168.0.0/16",
+	},
+}
+
+var conn1 = &connection.Connection{
+	Id:             "2",
+	NetworkService: "ucnf",
+	Mechanism: &connection.Mechanism{
+		Cls:  "LOCAL",
+		Type: "MEMIF",
+		Parameters: map[string]string{
+			"description": "NSM Endpoint",
+			"name":        "nsm2DoJzgpRd",
+			"netnsInode":  "4026534215",
+			"socketfile":  "nsm2DoJzgpRd/memif.sock",
+		},
+	},
+	Context: &connectioncontext.ConnectionContext{
+		IpContext: &connectioncontext.IPContext{
+			SrcIpAddr:     "192.168.22.1/30",
+			DstIpAddr:     "192.168.22.2/30",
+			SrcIpRequired: true,
+			DstIpRequired: true,
+			DstRoutes: []*connectioncontext.Route{
+				{Prefix: "192.168.0.0/16"},
+			},
+		},
+	},
+	Labels: map[string]string{
+		"":          "",
+		"namespace": "nsm-system",
+		"podName":   "helloworld-ucnf-6454c88c5f-pw98x",
+	},
+	Path: &connection.Path{
+		PathSegments: []*connection.PathSegment{
+			{
+				Name: "kind-2-control-plane",
+			},
+		},
+	},
+}
+
+//Changed DstIpAddr
+var conn2 = &connection.Connection{
+	Id:             "2",
+	NetworkService: "ucnf",
+	Mechanism: &connection.Mechanism{
+		Cls:  "LOCAL",
+		Type: "MEMIF",
+		Parameters: map[string]string{
+			"description": "NSM Endpoint",
+			"name":        "nsm2DoJzgpRd",
+			"netnsInode":  "4026534215",
+			"socketfile":  "nsm2DoJzgpRd/memif.sock",
+		},
+	},
+	Context: &connectioncontext.ConnectionContext{
+		IpContext: &connectioncontext.IPContext{
+			SrcIpAddr:     "192.168.22.1/30",
+			DstIpAddr:     "192.168.22.3/30",
+			SrcIpRequired: true,
+			DstIpRequired: true,
+			DstRoutes: []*connectioncontext.Route{
+				{Prefix: "192.168.0.0/16"},
+			},
+		},
+	},
+	Labels: map[string]string{
+		"":          "",
```

Comment:
Why to we need this "": ""? I find it a little bit strange, that's all

Reply:
Thanks for pointing this out! It was a mistake made by me. Will remove this line